### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by gnp665**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -45,22 +45,311 @@ If you need inspiration for what to write, take a look at previously approved st
...
+    }
+
+    std::cout << "The engine status is BAD!" << std::endl;
+    return false;
+}
+
+float Car::getMilesToEmpty(int gasTankSizeInGallons, float milesPerGallon, float gasPercentFull = 100.0f)
+{
+  return (gasTankSizeInGallons * milesPerGallon * gasPercentFull / 100);
+}
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -45,22 +45,311 @@ If you need inspiration for what to write, take a look at previously approved st
...
+
+int Studio::totalNumberMixes(int numClients, int mixesPerClient)
+{
+  return (numClients * mixesPerClient);
+}
+
+auto Studio::getTotalInvoiced(int numClients, int mixesPerClient, float hoursPerMix, float rate)
+{
+  return (numClients * mixesPerClient * hoursPerMix * rate);
+}
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -45,22 +45,311 @@ If you need inspiration for what to write, take a look at previously approved st
...
+    }
+
+    std::cout << "The engine status is BAD!" << std::endl;
+    return false;
+}
+
+float Plane::getMilesToEmpty(int gasTankSizeInGallons, float milesPerGallon, float gasPercentFull = 100.0f)
+{
+  return (gasTankSizeInGallons * milesPerGallon * gasPercentFull / 100);
+}
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -45,22 +45,311 @@ If you need inspiration for what to write, take a look at previously approved st
...
 
+struct Fleet
+{
+    Car myRedCar;
+    Plane myBluePlane;
+
+    Fleet();
+    ~Fleet();
+
+};
```
Comment (matkatmusic) [inline main.cpp]: I'm adding an instruction from the Project 5 Part 1 original instructions:
```
 4) add 2 member functions that use your member variables to each of these new UDTs
```
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -45,22 +45,311 @@ If you need inspiration for what to write, take a look at previously approved st
...
 
+struct MegaStudio
+{
+    Studio roomA;
+    Studio roomB;
+
+    MegaStudio();
+    ~MegaStudio();
+
+};
```
Comment (matkatmusic) [inline main.cpp]: I'm adding an instruction from the Project 5 Part 1 original instructions:
```
 4) add 2 member functions that use your member variables to each of these new UDTs
```
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Your code has formatting issues and produces a ton of warnings when compiled and run. 
About the warnings:
```
"Overshadow" means your function parameter's name is the same name as a class member variable.  
Change your function parameter's name so it is different from your member variable.
```
Re: unused param warning:  Use the parameter.
example:
```
void foo(int bar)
{
    //empty function.  bar is not used.  warning about unused param will appear.
}
//versus:
void foo(int bar)
{
    bar += 1;  //bar is used.  no warning will appear.
}
```
Fix the warnings and re-request a review

File: main.cpp (hunk 1 above)
Comment (gnp665) [reply main.cpp]: Fixed

File: main.cpp (hunk 2 above)
Comment (gnp665) [reply main.cpp]: Fixed

File: main.cpp (hunk 3 above)
Comment (gnp665) [reply main.cpp]: Fixed

File: main.cpp (hunk 4 above)
Comment (gnp665) [reply main.cpp]: Added 2 member functions

File: main.cpp (hunk 5 above)
Comment (gnp665) [reply main.cpp]: Added 2 member functions

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**